packages/block-editor/src/components/dimension-control/index.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { isFunction } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	Icon,
+	SelectControl,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+import {
+	Fragment,
+} from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import sizesTable, { findSizeBySlug } from './sizes';
+
+export function DimensionControl( props ) {
+	const { label, icon, sizes = sizesTable, currentSize, onSpacingChange, onReset, className = '' } = props;
+
+	/**
+	 * Determines the size from the size slug (eg: `medium`)
+	 * and decides whether to call the change or reset callback
+	 * handlers
+	 * @param  {string} val the DOMEvent event.target
+	 * @return {void}
+	 */
+	const onChangeSpacingSize = ( val ) => {
```

Comment:
I guess we can wrap the function definitions with useCallback to avoid creating a new function on each rerender.

Reply:
Can I check why you are suggesting we use `useCallback`  here? My understanding that it was effectively to "memoize" computationally expensive functions and/or to ensure referential equality.

I assume you're coming at this with a perf lense? If so do you feel that `onChangeSpacingSize` is sufficiently intensive to warrant a `useCallback`?

If I'm missed something I'd be open to an explanation - I'm always looking to improve my understanding, especially in the field of hooks!

Reply:
Hi @getdave, 

> My understanding that it was effectively to "memoize" computationally expensive functions and/or to ensure referential equality.

We share the understanding. I'm suggesting use callback to ensure referential equality.
Without use callback, we will pass new onChange value to SelectControl on each render. This makes SelectControl need to rerender and makes React need to update the dom events to reference a new callback. All of this seems wasted work and use callback makes it simple to avoid this wasted work.

Reply:
🙇 Thanks. I needed to grok that. Much appreciated.